design/0003-isolated-integration-tests.md

```diff
@@ -0,0 +1,324 @@
+- Contribution Name: Isolated Integration Tests
+- Implementation Owner: @kleimkuhler, @alpeb
+- Start Date: 2020-03-18
+- Target Date: 2020-04-03
+- RFC PR: [linkerd/rfc#12](https://github.com/linkerd/rfc/pull/12)
+- Linkerd Issue:
+  [linkerd/linkerd2#4190](https://github.com/linkerd/linkerd2/issues/4190)
+- Reviewers: @grampelberg @olix0r
+
+# Summary
+
+[summary]: #summary
+
+Linkerd's integration testing process will provide developers a way to test
+basic Linkerd installations as well as more complex ones, such as
+multi-cluster installations and cluster failure scenarios.
+
+A better UX will be available for running all or one of the integration tests.
+Additionally, running the tests will be less reliant on existing cluster
+state.
+
+With these changes to Linkerd's integration testing process, developers will
+be more equip to run Linkerd tests locally without surprises. Adding new tests
+for features that require more complex cluster conditions will be possible.
+
+# Problem Statement (Step 1)
+
+[problem-statement]: #problem-statement
+
+The Linkerd integration test suite is extensive and effective at catching
+bugs, but it does not provide a very positive UX experience to new or existing
+Linkerd developers.
+
+It is also unable to provide many options for testing complex cluster
+scenarios such as multi-cluster installations and cluster failure scenarios.
+
+The are several reasons for these existing problems and each should be
+addressed by the change this RFC is proposing.
+
+### UX
```

Comment:
I think it's also worth calling out the dangers of running tests with the wrong kube context.